Frameworks/Foundation/NSString.mm

```diff
@@ -740,13 +743,11 @@ - (NSString*)description {
  @Status Interoperable
 */
 - (BOOL)isEqualToString:(NSString*)compStr {
-    RETURN_FALSE_IF(!compStr);
-
-    if ([compStr isKindOfClass:[NSString class]]) {
```

Comment:
>isKindOfClass [](start = 17, length = 13)

We can't use the short path for custom classes (such as NSStringOnHStting).  Why is this removed?